### DISCUSSION TITLE
Traceback when Cythonize and strip the binary

### DISCUSSION THREAD

**Discussion by ArduinoHocam** [pypa/setuptools#4155] [Q&A]
Hi,

I have a python3 codes and using the setuptools, cythonization is performed with the settings in the setup.py.

A brief process I have is:
- `python3 setup.py build_ext -j 12 bdist_wheel`
- `wheel unpack <mod> -d <dir>`
- find *.so
- `x86_64-linux-gnu-strip -s <.so>`.
-` wheel pack <pk_dir> -d <dir>`

while compiling I see the following output:
`x86_64-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -ffile-prefix-map=/build/python3.9-RNBry6/python3.9-3.9.2=. -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -fPIC -I/usr/include/python3.9 -c foobar/foo.c -o build/temp.linux-x86_64-cpython-39/foobar/foo.o -Os -s -DCYTHON_WITHOUT_ASSERTIONS`

My first question is, is there a way to remove these default flags that comes from `CCLFAGS` I guess? Because I can only extend the flags in the `setup.py` using `extra_compile_args` argument.

Second is:

In the platform, I invoke the compiled code as follows:

`python3 <dummy_file_contains_import_for_so_file>`
and if a code crashes, I can easily see the line numbers, function names etc.
How is this possible? because if I check the .so file with `nm` tool, it says no symbols. because I have used strip tool.
I don't get it how python interpreter gets the all information and shows it.

**Comment by webknjaz** [comment]
First of all, calling `setup.py` on command line is deprecated and have been discouraged for years. Use pypa/build to make dists.

Also, this sounds like a Cython question, rather than a setuptools one. You probably have to study their docs instead.

**Reply by ArduinoHocam** [reply]
Hi @webknjaz ,
Thank you for your answer. The part where the deprecated part is used not a big problem now, I am quite surprised about the question(2nd) I have asked. Because does not makes sense.

Since I was using `setuptools` I thought this should be the place for question.

**Comment by ArduinoHocam** [comment]
The answer I got in the Cython group is that:
"Cython embeds the line numbers, file names and function names in the compiled .so file. These are not symbols that you can strip - they're just ordinary C strings/numbers. These are used to look up the lines in the .pyx file if it's present, otherwise they're just shown without the source."
So closing.
